### PR TITLE
Update find and storage documentation

### LINKED ISSUE #47: Update Developer Guide
On Architecture Diagrams and EditCommands implementations.

### PR DESCRIPTION
**Pull request by daniellau88**
- Update documentation for find command
- Update documentation for storage architechture

[FindCommandSequenceDiagram](https://drive.google.com/file/d/1Sk68JaLV_d5XFZOsi_OA6pz4qRxAN5FR/view?usp=sharing)
[FindPredicateClassDiagram](https://drive.google.com/file/d/1nQE77SHkWd0b_nc4YMTuxFaIUrc88kQl/view?usp=sharing)
[StorageClassDiagram](https://drive.google.com/file/d/1dLYIJliOL6P9XBTEm07uhktOQEw6HHzE/view?usp=sharing)

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -25,12 +25,59 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
...
+
+The `Storage` component,
+* can save `UserPref` objects in json format and read it back.
+* can save the address book data in json format and read it back.
+
 ### Common classes
 
+Classes used by multiple components are in the `seedu.addressbook.commons` package.
+
+
```
Comment (weixue123) [inline docs/DeveloperGuide.md]: Do add the --- divider and remove the [In Progress]

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -25,12 +25,59 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
...
+sum of the relevance of each predicate.
+
+The following shows the structure of the `FieldPredicate` classes which will be used to
+filter and sort the different lists.
+
+![Architecture of the Find Command](images/find/FindPredicateClassDiagram.png)
+
+The diagram below shows how the `FindCommand`, specifically the `FindCheeseCommand` works:
+
+![Sequence of Find Command](images/find/FindCheeseCommandSequenceDiagram.png)
```
Comment (weixue123) [inline docs/DeveloperGuide.md:201]: Is the diagram missing some return arrows?

Also, I think we should standardize, as a group, whether to include the argument strings in our sequence diagram

**Review by weixue123 — APPROVED** [review]
Diagrams looks good! Can merge after the small adjustment

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (daniellau88) [reply docs/DeveloperGuide.md:201]: I only didn't include the arrows for updateFilteredCheeseList (it returns void), the rest are included

**Pull request merged**